apps/builder/app/builder/features/command-panel/token-group.tsx

```diff
@@ -0,0 +1,224 @@
+import { useState } from "react";
+import { matchSorter } from "match-sorter";
+import { computed } from "nanostores";
+import { useStore } from "@nanostores/react";
+import {
+  CommandGroup,
+  CommandGroupHeading,
+  CommandInput,
+  CommandItem,
+  CommandList,
+  Flex,
+  ScrollArea,
+  Text,
+  toast,
+  useSelectedAction,
+} from "@webstudio-is/design-system";
+import type { Instance, Instances, StyleSource } from "@webstudio-is/sdk";
+import {
+  $instances,
+  $pages,
+  $registeredComponentMetas,
+  $selectedStyleSources,
+  $styleSources,
+  $styleSourceSelections,
+} from "~/shared/nano-states";
+import { getInstanceLabel } from "~/shared/instance-utils";
+import type { InstanceSelector } from "~/shared/tree-utils";
+import { $awareness } from "~/shared/awareness";
+import { $commandContent, closeCommandPanel } from "./command-state";
+
+export type TokenOption = {
+  terms: string[];
+  type: "token";
+  token: Extract<StyleSource, { type: "token" }>;
+  usages: number;
+};
+
+const $styleSourceUsages = computed(
```

Comment:
I understand that we use computed everywhere else, but search is a different case, you don't need to compute this stuff on every change, because its only needed when search gets open, so this computation is wasted as long as the search is not open

Reply:
IMO search data should be gathered only when search interface opens + all data should be normalized to one format, so that the search works the same way for all types of data

Reply:
Like a sqlite database that boots when you need it

Reply:
Computed are invoked lazily when something is subscribed to them

Reply:
This is why all stored are subscribed inside of dialog

Reply:
weird, I thought computed subscribes all the time

Reply:
It would completely kill performance

Reply:
when I use $something.get() it doesn't know if there is a subscription, will get invoke computed as well then?

Reply:
One time subscription to recompute all stuff when .get is invoked.